features/storage/TESTS/blockdevice/general_block_device/main.cpp

```diff
@@ -209,6 +213,196 @@ void test_multi_threads()
     TEST_ASSERT_EQUAL(0, err);
 }
 
+void test_get_erase_value()
+{
+    utest_printf("\nTest BlockDevice::get_erase_value()..\n");
+
+    // Test flow:
+    //  1. Write data to selected region
+    //    - Known starting point
+    //  2. Erase selected region
+    //  3. Read erased region and compare with get_erase_value()
+
+    BlockDevice *block_device = BlockDevice::get_default_instance();
+    TEST_SKIP_UNLESS_MESSAGE(block_device != NULL, "\nno block device found.\n");
+
+    int err = block_device->init();
+    TEST_ASSERT_EQUAL(0, err);
+
+    // Check erase value
+    int erase_value_int = block_device->get_erase_value();
+    utest_printf("\nblock_device->get_erase_value()=%d", erase_value_int);
+    TEST_SKIP_UNLESS_MESSAGE(erase_value_int >= 0, "\nerase value is negative which means the erase value is unknown\n");
+
+    // Assuming that get_erase_value() returns byte value as documentation mentions
+    // "If get_erase_value() returns a non-negative byte value" for unknown case.
+    TEST_ASSERT(erase_value_int <= 255);
+    uint8_t erase_value = (uint8_t)erase_value_int;
+
+    // Determine starting address
+    bd_size_t data_buf_size = block_device->get_erase_size();
+    TEST_ASSERT(data_buf_size > 0);
+    bd_addr_t start_address = rand();       // low 32 bytes
+    start_address += (uint64_t)rand() << 8; // high 32 bytes
```

Comment:
AFAIU, The remarks here don't match the code:
rand() will return a 15bits "random" number, and adding shifting it another "random" 15bits number shifted by 8bits  (<< 8) will make it together a 23bits random number. 
are you trying to generate a random start_address between 0 and (block_device->size() - data_buf_size) ?

Reply:
edit: Unlike what I wrote above, which is for visual, in GCC rand() will return a 32bit random number. So, if you want to make it 64 bit random you need to shift the second part by 32bits ( <<32).

I'm not sure what's the highest value of rand() on ARM or IAR compiler

Reply:
Oops, the shift is definitely wrong, good catch.